mempool/cache.go

```diff
@@ -18,14 +18,14 @@ type TxCache interface {
 
 	// Push adds the given raw transaction to the cache and returns true if it was
 	// newly added. Otherwise, it returns false.
-	Push(tx *types.CachedTx) bool
+	Push(key types.TxKey) bool
```

Comment:
why is `CachedTx` removed?

Reply:
because that's how it was done in v0.38 which I noticed after having to resolve merge conflicts. Also, the function signature is cachedTx but it was only saving the key. https://github.com/celestiaorg/celestia-core/blob/5585246da4e6d203adced5a5548275c4c78dc783/mempool/cache.go#L71

Reply:
yes, the purpose of cachedTx is to save the key/hash without having to rehash it every time

Reply:
but why can't you directly pass the key through cachedTx from the mempool? this allows us to use the same cache for priority and cat